src/components/pages/main/components/book/Book.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from "react";
+import { IBook } from "../../../../../interfaces";
+import "./Book.css";
+import { Link } from "react-router-dom";
+import { ThemeContext } from "../../../../../ThemeContext";
+
+interface BookProps {
+  book: IBook;
+  currentPage: number;
+}
+
+const Book: React.FC<BookProps> = ({ book, currentPage }) => {
```

Comment:
```suggestion
type Props = {
  book: IBook;
  currentPage: number;
}

const Book: FC<Props> = ({ book, currentPage }) => {
```
при функциональных компонентах чаще юзают типы. и если ты описываешь пропсы компонента, в модуле этого компонента, то норм просто писать пропс.

плюс из реакто обычно все же импортирует деструктуризацией, типа FC вместо React.FC, useEffect вместо React.useEffect